src/search-backend/metadata-search-backend.ts

```diff
@@ -26,6 +22,10 @@ export class MetadataSearchBackend extends BaseSearchBackend {
   async performSearch(
     params: SearchParams
   ): Promise<Result<any, SearchServiceError>> {
+    if (this.debuggingEnabled && params.debugging == null) {
```

Comment:
re: `params.debugging == null`

- we normally use stricter === comparator.
- can `!params.debugging` false work?
- also, how can I find the type? can we do `params.debugging: boolean` instead of `params.debugging: true | false | null`?

Reply:
I used coercing `==` here to compare against both nullish types together (null / undefined). But I suppose this could be simplified to `params.debugging == undefined` since `null` shouldn't be a valid type for `params.debugging`.

The intent is to allow the per-request `debugging` param to override the search backend's `debuggingEnabled` setting for that request only, so setting `debugging: false` is meant to fail this condition. Not setting the per-request param should defer to the backend prop.